Sass changes

osk opened the pull request:
Changes to default Sass:

* Add scaling css var and calc to responsive-font to deal with wide pages
* Move hardcoded values to variables
* Deprecate `viewport` mixin via `@warn`
* Update `$min-mobile` breakpoint to 420 from 480
* Update `$page-width` to `1290px`
* Rename (and deprecate old) grid mixins to `grid-col` etc
* Add heading() subheading and copy() mixinstemplates

osk committed:
Move grid constants into vars, rename and deprecate grid mixins

Variables:
$grid-column-count and $grid-baseline
Deprecate:
make-row (now grid-row)
make-col (now grid-col)
make-offset-left (now grid-offset-left)
make-offset-right (now grid-offset-right)

osk committed:
Deprecate viewport mixin, name breakpoints

osk committed:
Move all hardcoded values into variables

osk committed:
Update mobile breakpoint from 480 to 420

osk committed:
Update page-width to 1290, the ueno grid width

osk committed:
Changes gathered from other projects

osk committed:
Add warning to responsive-font if min-size >= size

osk committed:
Bug fixes

osk committed:
Add heading() subheading and copy() mixins templates

organdonor47 commented on shared/styles/mixins.scss:
```diff
@@ -83,7 +97,7 @@
   padding: 0 $container-gutter-mobile;
   max-width: $page-width + $container-gutter-mobile * 1;
```
this mobile stuff in `container` doesn't really do anything; `$page-width` is always going to be greater than mobile viewports anyway and I am not sure why there are differences between the base size (`$page-width` + ` $container-gutter-mobile * 1`) and min-mobile anyway? 

also, there may be instances where we need to override top & bottom padding / margins, so this would be over-written at the next mq: 

```
&__container {
    @include container;

    padding-top: percentage(50px/$page-limit);
    // overwritten at tablet to 0;
}
```

could left & right be explicitly declared as long-hand values rather than short-hand?

Like: 

```
@mixin container() {
  margin-left: auto;
  margin-right: auto;
  padding-left: $container-gutter-mobile;
  padding-right: $container-gutter-mobile;

  @media (min-width: $min-tablet) {
    padding-left: $container-gutter-tablet;
    padding-right: $container-gutter-tablet;

    max-width: $page-width + $container-gutter-tablet * 2;
  }

  @media (min-width: $min-desktop) {
    padding-left: $container-gutter-desktop;
    padding-right: $container-gutter-desktop;

    max-width: $page-width + $container-gutter-desktop * 2;
  }
}
```

osk replied:
It should definitely not use shorthand for padding.

organdonor47 reviewed (commented):
maybe tidy that `container` mixin? added a comment on that.

also, we could add that weird dimensions breakpoint fix that we talked about & is in this PR - https://github.com/ueno-llc/ueno-www/pull/342 -
(will need to be documented & maybe even commented out by default?)

geirlandro commented on shared/styles/grid.scss:
```diff
@@ -39,10 +39,31 @@
+@mixin make-row($direction: ltr, $align: stretch, $justify: flex-start, $grid-gutter: $gutter, $wrap: wrap) {
+  @warn 'make-row mixin is deprecated, use grid-row';
+  @include grid-row($direction, $align, $justify, $grid-gutter, $wrap);
+}
```
You can do:
```scss
@mixin make-row($args...) {
  ...
  @include grid-row($args...);
}
```

osk commented:
Yeah, I had the scale thingie in mind. Has it been reviewed? Or should I just port it here and add to a page for easier review?

organdonor47 commented:
yeah @osk maybe add in from that PR & we can test here instead -- not sure it needs to be implemented for ueno.co anyway (or at least there is no urgent need)

bjarnif commented:
It would be nice to add these to html {} in base.scss 
-webkit-font-smoothing: antialiased; // stylelint-disable-line
-moz-osx-font-smoothing: grayscale; // stylelint-disable-line

osk committed:
Use sass variable arguments syntax for deprecated mixins

osk committed:
Add vendor prefixed font smoothing to base

osk committed:
Use longhand for media query padding in container mixin

osk committed:
Lint and decrease usage of stylelint-disable-line

osk commented:
I've updated according to comments.

@organdonor47 I'll add the scale experiment in a new branch after this one is merged.

j2em6h commented on shared/styles/grid.scss:
```diff
@@ -39,10 +39,31 @@
 }
 
-@mixin make-offset-right($offset: percentage(1/12)) {
+@mixin grid-offset-right($offset: percentage(1/$grid-column-count)) {
```
Could the default `$offset:` just be `1` on lines `42` and `46`? Because the `percentage($number / $grid-column-count)` is already getting applied via `convertify($offset)`?

osk replied:
Yeah, since that's the expected behaviour when you're using it

j2em6h commented on shared/styles/mixins.scss:
```diff
@@ -83,17 +97,69 @@
+  @include reset-heading;
+  @include responsive-font($min, $max);
+
+  @warn 'Please implement for project';
```
What does `implement for project` mean?

osk replied:
These three mixins are included for those that want them. They have some settings that need to be changed for the project you're using them for, so this should trigger you to change them. But the wording is bad.

`Please set values in this mixin for your project.`?

j2em6h replied:
Hmmm, are these really that different than any other piece of the starter kit? Anything _could_ be customized for a project if it needs to be, right?

osk replied:
True, but these values are arbitrary, like the `58` for margin, other values are based on `$gutter` or have some logic behind them.

j2em6h replied:
It doesn't _have_ to be arbitrary, it could be based off something. Maybe even `($gutter * 2)`; or some other new variable.

organdonor47 replied:
@k80 @osk i think if we add a variable there it could be mistaken for an integral non-editable part of the mixin though. an arbitrary value might look a bit messy, but at least it indicates a value you can edit? maybe we could have warnings in there to update those values and just have some kind of obvious placeholder value?

j2em6h replied:
`123456789px` would be obvious… and totally break the design.

osk committed:
Add scaling css var and calc to responsive-font to deal with wide pages

First version with simple calculation with fallback

osk committed:
Add Heading component to test and showcase responsive-font

j2em6h commented on shared/styles/mixins.scss:
```diff
@@ -83,17 +97,69 @@
   padding: 0 $container-gutter-mobile;
```
Should this also be long hand?

osk replied:
Hmm, I'm not sure. Is there a case where we need to explicitly force it to be `0`? @organdonor47 @mdmagnusson ?

organdonor47 replied:
@osk @k80 yeh it should i think. no reason why it needs to be 0 top & bottom

osk commented:
On second thought, added a first version of scaling to `responsive-font`

j2em6h commented on shared/styles/mixins.scss:
```diff
@@ -83,17 +97,69 @@
+  font-family: $font-family;
+  font-weight: normal;
+  letter-spacing: 0;
+  line-height: (58/52);
```
Is this `/52` supposed to be `/$max`? It is in the other 2.

osk committed:
Update `subheading` mixin

osk committed:
Update after review

grid-offset-* default value 1 instead of 1/$grid-column-count
container mixin uses padding longhand
heading, subheading and copy mixins don't warn, just have explicitly
arbitrary values

osk committed:
Merge branch 'feature/scss-changes' of github.com:ueno-llc/starter-kit-universally into feature/scss-changes

osk commented:
@k80 @organdonor47 
I removed the `@warn` from the mixins and set explicit `$arbitrary-*` variables instead it would always be changed, right? 😊

osk commented:
I'm merging this later today if there are no objections

organdonor47 commented:
i think there is probably a more sensible solution to this down the line, but this is as good as any other alternative I can think of!
